scripts/aggregate-themes/aggregate-themes.js

```diff
@@ -0,0 +1,150 @@
+const fs = require('fs');
+const glob = require('glob');
+const path = require('path');
+const Logger = require('../utils/logger');
+
+const CONFIG = 'theme.config.js';
+const NODE_MODULES = 'node_modules/';
+const OUTPUT = 'aggregated-themes.js';
+const OUTPUT_PATH = path.resolve(process.cwd());
+const DISCLAIMER = fs.readFileSync(path.resolve(__dirname, 'disclaimer.txt'), 'utf8');
+
+/**
+ * Aggregates theme assets into a single file.
+ *
+ * By default the theme will recursively search all dependencies.
+ * If a theme directory contains a root-theme.scss only that single file will be aggregated.
+ */
+class ThemeAggregator {
+  /**
+   * Aggregates theme assets.
+   * @returns {string|null} - The output path of the aggregated theme file. Null if not generated.
+   */
+  static aggregate() {
+    const defaultConfig = path.resolve(process.cwd(), CONFIG);
+
+    if (fs.existsSync(defaultConfig)) {
+      // eslint-disable-next-line global-require, import/no-dynamic-require
+      return ThemeAggregator.aggregateTheme(require(defaultConfig));
+    }
+
+    return null;
+  }
+
+  /**
+   * Aggregates theme assets.
+   * @param {Object} options - The aggregation options.
+   * @returns {string} - The output path of the aggregated theme file.
+   */
+  static aggregateTheme(options = {}) {
+    ThemeAggregator.validate(options);
+
+    const { theme } = options;
+
+    const dirs = ThemeAggregator.find(`**/themes/${theme}/`, options);
+    const assets = ThemeAggregator.filter(dirs, options);
+
+    if (fs.existsSync(`${NODE_MODULES}${theme}/`)) {
+      assets.unshift(theme);
```

Comment:
why are we removing the `node_modules/theme` from the assets list?

Reply:
We're not removing it from the list. Unshift appends it to the beginning of the list.

Reply:
🤦‍♀️